tf2_ros_py/tf2_ros/buffer.py

```diff
@@ -84,8 +85,16 @@ def __init__(
         self._callbacks_to_remove: List[Callable[[], None]] = []
         self._callbacks_lock = threading.RLock()
 
+        self.clock = rclpy.clock.Clock()
         if node is not None:
             self.srv = node.create_service(FrameGraph, 'tf2_frames', self.__get_frames)
+            self.clock = node.get_clock()
```

Comment:
I'm not a huge fan of creating the clock object, and then potentially immediately switching it out.  That just leads to unnecessary objects that the garbage collector will eventually have to clean up.  Instead, I'd suggest:

```suggestion
        if node is not None:
            self.srv = node.create_service(FrameGraph, 'tf2_frames', self.__get_frames)
            self.clock = node.get_clock()
        else:
            self.clock = rclpy.clock.Clock()
```

Reply:
The tests fails with this version of the code. At the moment I am unsure why.

Reply:
I'm pretty sure it is unrelated to this.  I kicked off the tests again.